curriculum/challenges/english/01-responsive-web-design/basic-html-and-html5/inpaint.html

```diff
@@ -0,0 +1,113 @@
+<html>
+<head>
+<meta charset="utf-8">
+<title>CVdrawing</title>
+</head>
+<body>
+<p id="status">OpenCV.js is loading...</p>
+<canvas id="canvas" width="640" height="426">
+</canvas>
+<img id="src" src="bird.jpg" style="display:none"/>
+<input type="button" id="myButton" value="Test" disabled=true
+onclick="inpaint()"/>
+
+<script type="text/javascript">
+
+var checker = false;
+var src;
+var srcmat;
+var dst;
+var dstmat;
+var mask;
+var lastX;
+var lastY;
+var x;
+var y;
+
+window.onload = function()
+{
+context = document.getElementById("canvas").getContext("2d");
+img = document.getElementById("src");
+context.drawImage(img, 0, 0);
+};
+
+document.getElementById("canvas").onmousedown = function(e)
+{
+  x = e.pageX - this.offsetLeft;
+  y = e.pageY - this.offsetTop;
+  if (lastX == undefined)
+  {
+      lastX = x;
+      lastY = y;
+  }
+};
+
+document.getElementById("canvas").onmousemove = function(e)
+{
+  if(lastX != undefined)
+  {
+    cv.line(src, new cv.Point(lastX, lastY), new cv.Point(x, y), new cv.Scalar(255, 255, 255, 255), 2);
+    cv.line(mask, new cv.Point(lastX, lastY), new cv.Point(x, y), new cv.Scalar(255, 255, 255, 255), 2);
+    lastX = x;
+    lastY = y;
+    x = e.pageX - this.offsetLeft;
+    y = e.pageY - this.offsetTop;
+    cv.imshow("canvas", src);
+  }
+};
+
+document.getElementById("canvas").onmouseup = function(e)
+{
+  if (lastX != undefined)
+  {
+    lastX = undefined;
+    lastY = undefined;
+  }
+};
+
+function init()
+{
+  if (checker == true)
+  {
+    src = cv.imread("src");;
+    dst = src.clone();
+    let cols = src.size().width;
+    let rows = src.size().height;
+    let ch = src.channels();
+    srcmat = new cv.Mat();
+    mask = new cv.Mat(426, 640, cv.CV_8UC1, new cv.Scalar(0, 0, 0, 255));;
```

Comment:
mask should be the same size as src and dst.

Reply:
Solved via:

Reply:
Solved via:
img = document.getElementById("src");
context.canvas.width = img.width;
context.canvas.height = img.height;